avivator/src/components/Viewer.jsx

```diff
@@ -41,6 +41,12 @@ const Viewer = () => {
     onViewportLoad,
     useFixedAxis
   } = useImageSettingsStore();
+
+  const onViewStateChange = ({ viewState: { zoom } }) => {
+    const z = Math.min(Math.max(Math.round(-zoom), 0), loader.length - 1);
+    useImageSettingsStore.setState({ pyramidResolution: z });
+  };
```

Comment:
This should probably be debounced.  It seems to make the demo very slow.  Are you seeing this too?

Reply:
Hmmm, I haven't used `zustand` before, but reading though the docs, I think the issue is that `useImageSettingsStore` holds a lot of state, and the use of stores in Avivator doesn't granularly "pick" state to listen for, e.g.

```javascript
const { resolution } = useImageSettingsStore(); // triggers rerender every time _any_ state is changes in this store
// vs.
const resolution = useImageSettingsStore(store => store.resolution); // rerender only when resolution changes
````

So because this state update is called frequently, it bogs down the UI. I moved this state to its own store because changing all the code to use the latter "picking" pattern is beyond the scope of this PR, and it works much more smoothly now.

Reply:
Another thing I noticed is that many of the stores have a custom setter for merging state. `zustand` also implements this natively..

```javascript
// current 
function Component() {
  const { setViewerState } = useViewerStore();
  setViewerState({ source, isNoImageUrlSnackbarOn });
  / ... 
}

// built-in
function Component() {
  userViewerStore.setState({ source, isNoImageUrlSnackbarOn });
  / ...
}
```

Again, I think the latter pattern is preferred because the former is listening for changes and will trigger a rerender if any of the  ViewerStore changes, whereas the latter is an emitter only (no listener).